[TensorPipe] Add extensive logging

lw opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#39781 [TensorPipe] Add extensive logging**

Use a new feature of TensorPipe where a pipe can tell you the name of the remote endpoint, in order to make the logging messages more informative: whenever there is a failure on a pipe, say which worker this was to/from, and the ID of the message involved.

Also, add plenty of verbose logging, to help with debugging. This is off by default, but can be enabled by setting the `GLOG_v` env var to a value of 1 or higher.

Differential Revision: [D21973150](https://our.internmc.facebook.com/intern/diff/D21973150/)

lw committed:
[TensorPipe] Add extensive logging

Use a new feature of TensorPipe where a pipe can tell you the name of the remote endpoint, in order to make the logging messages more informative: whenever there is a failure on a pipe, say which worker this was to/from, and the ID of the message involved.

Also, add plenty of verbose logging, to help with debugging. This is off by default, but can be enabled by setting the `GLOG_v` env var to a value of 1 or higher.

Differential Revision: [D21973150](https://our.internmc.facebook.com/intern/diff/D21973150/)

[ghstack-poisoned]

lw committed:
Update on "[TensorPipe] Add extensive logging"


Use a new feature of TensorPipe where a pipe can tell you the name of the remote endpoint, in order to make the logging messages more informative: whenever there is a failure on a pipe, say which worker this was to/from, and the ID of the message involved.

Also, add plenty of verbose logging, to help with debugging. This is off by default, but can be enabled by setting the `GLOG_v` env var to a value of 1 or higher.

Differential Revision: [D21973150](https://our.internmc.facebook.com/intern/diff/D21973150/)

[ghstack-poisoned]

osalpekar reviewed (approved):
Thanks - this will be really useful when debugging!

lw commented on torch/csrc/distributed/rpc/tensorpipe_agent.cpp:
```diff
@@ -127,13 +127,17 @@ TensorPipeAgent::TensorPipeAgent(
 }
 
 TensorPipeAgent::~TensorPipeAgent() {
+  VLOG(1) << "RPC agent for " << workerInfo_.name_ << " is being destroyed";
```
I added it because at some point I was investigating shutdown-related issues (see also #39663). As the current RPC agent is stored in shared_ptrs the moment it's destructed isn't necessarily easy to predict, as any user holding on their shared_ptr can delay it. I think it may make sense to have this information when debugging that type of issues.

lw commented on torch/csrc/distributed/rpc/tensorpipe_agent.cpp:
```diff
@@ -591,31 +638,45 @@ void TensorPipeAgent::join() {
 void TensorPipeAgent::shutdownImpl() {
-  LOG(INFO) << "Shutting down TensorPipeAgent on node "
-            << RpcAgent::getWorkerInfo().name_;
+  VLOG(1) << "RPC agent for " << workerInfo_.name_ << " is shutting down";
```
Yeah, I saw it was `LOG(INFO)` and was wondering why. I'm happy to switch it back if that was intentional, for consistency. I am however wondering what the take of PyTorch is on having a library print log messages even during normal operation. (In TensorPipe, for example, we avoid that completely: we only log warnings/errors by default, and verbose logging must be explicitly enabled).

lw committed:
Update on "[TensorPipe] Add extensive logging"


Use a new feature of TensorPipe where a pipe can tell you the name of the remote endpoint, in order to make the logging messages more informative: whenever there is a failure on a pipe, say which worker this was to/from, and the ID of the message involved.

Also, add plenty of verbose logging, to help with debugging. This is off by default, but can be enabled by setting the `GLOG_v` env var to a value of 1 or higher.

Differential Revision: [D21973150](https://our.internmc.facebook.com/intern/diff/D21973150/)

[ghstack-poisoned]
